include/pstore/mcrepo/compilation.hpp

```diff
@@ -53,11 +53,15 @@ namespace pstore {
     namespace repo {
 
 #define PSTORE_REPO_LINKAGE_TYPES                                                                  \
-    X (append)                                                                                     \
-    X (common)                                                                                     \
-    X (external)                                                                                   \
-    X (internal)                                                                                   \
-    X (linkonce)
+    X (append_linkage)                                                                             \
```

Comment:
Do we need to have the word “linkage” added to the end of all of these? They're defined in a enum class named linkage_type so isn’t this redundant?

Reply:
The "private" is a keyword.  I had to borrow the format of the LLVM linkage type and add the `_linkage` suffix.

Reply:
Maybe use a different word? The LLVM sources define `PrivateLinkage` as "Like Internal, but omit from symbol table"; perhaps `internal_no_symbol`?

Reply:
Good point. I have updated the patch following your advice.